plugins/jobs/src/js/types/Job.ts

```diff
@@ -0,0 +1,200 @@
+import {
+  JobDetailResponse as MetronomeJobDetailResponse,
+  JobStatus as MetronomeJobStatus
+} from "#SRC/js/events/MetronomeClient";
+import {
+  JobRunConnection,
+  JobRunConnectionTypeResolver,
+  JobRunConnectionSchema
+} from "#PLUGINS/jobs/src/js/types/JobRunConnection";
+import {
+  JobHistorySummary,
+  JobHistorySummaryTypeResolver,
+  JobHistorySummarySchema
+} from "#PLUGINS/jobs/src/js/types/JobHistorySummary";
+import {
+  JobRunStatusSummary,
+  JobRunStatusSummaryTypeResolver,
+  JobRunStatusSummarySchema
+} from "#PLUGINS/jobs/src/js/types/JobRunStatusSummary";
+import {
+  JobScheduleConnection,
+  JobScheduleConnectionTypeResolver,
+  JobScheduleConnectionSchema
+} from "#PLUGINS/jobs/src/js/types/JobScheduleConnection";
+import {
+  JobStatus,
+  JobStatusSchema
+} from "#PLUGINS/jobs/src/js/types/JobStatus";
+import { JobHistoryRun } from "#PLUGINS/jobs/src/js/types/JobHistoryRun";
+import {
+  JobDockerTypeResolver,
+  JobDocker,
+  JobDockerSchema
+} from "#PLUGINS/jobs/src/js/types/JobDocker";
+import { JobLabel, LabelSchema } from "#PLUGINS/jobs/src/js/types/JobLabel";
+import { cleanJobJSON } from "#SRC/js/utils/CleanJSONUtil";
+export interface Job {
+  activeRuns: JobRunConnection;
+  command: string;
+  cpus: number;
+  description: string;
+  disk: number;
+  docker: JobDocker | null;
+  id: string;
+  jobRuns: JobRunConnection;
+  json: string;
+  labels: JobLabel[];
+  lastRunsSummary: JobHistorySummary;
+  lastRunStatus: JobRunStatusSummary;
+  mem: number;
+  name: string;
+  schedules: JobScheduleConnection;
+  scheduleStatus: JobStatus;
+}
+
+export const JobSchema = `
+${JobRunConnectionSchema}
+${LabelSchema}
+${JobHistorySummarySchema}
+${JobRunStatusSummarySchema}
+${JobStatusSchema}
+${JobDockerSchema}
+${JobScheduleConnectionSchema}
+
+type Job {
+  activeRuns: JobRunConnection!
+  command: String!
+  cpus: Float!
+  description: String!
+  disk: Float!
+  docker: Docker
+  id: ID!
+  jobRuns: JobRunConnection!
+  json: String!
+  labels: [Label]!
+  lastRunsSummary: JobHistorySummary!
+  lastRunStatus: JobRunStatusSummary!
+  mem: Int!
+  name: String!
+  schedules: ScheduleConnection!
+  scheduleStatus: JobStatus!
+}
+`;
+
+export function JobTypeResolver(response: MetronomeJobDetailResponse): Job {
+  return {
+    id: JobFieldResolvers.id(response),
```

Comment:
Why do we need so many function calls?

Reply:
Okay got it, but why are the functions part of an object? I am scared that we at some point remove a resolver but we forget to remove the function for the resolver from the object and then we carry unused code around.

Reply:
@DanielMSchmidt :trollface:

Reply:
@Poltergeist 

a) Code is always subject to change, why is this here different then somewhere else?
b) These will be inlined later to a place where they are right beside their schema. Literally 10 lines away. The will also not be called by hand but by graphql. If three of us overlook this while coding and reviewing we deserve to have stray code around. 

I honestly think you have a point here in general and we should file a jira to research tooling to find dead code, but I think this is unrelated to this PR

Reply:
👍 That is why this is not blocking 👍